tests/integration/golang/mlflow/run/log_batch_test.go

```diff
@@ -397,10 +424,20 @@ func (s *LogBatchTestSuite) TestMetrics_Ok() {
 			s.Empty(resp)
 
 			// make sure that `iter` and `last_iter` for each metric has been updated correctly.
-			for key, iteration := range tt.latestMetricIteration {
-				lastMetric, err := s.MetricFixtures.GetLatestMetricByKey(context.Background(), key)
-				s.Require().Nil(err)
-				s.Equal(iteration, lastMetric.LastIter)
+			if len(tt.latestMetricIteration) > 0 {
+				for key, iteration := range tt.latestMetricIteration {
+					lastMetric, err := s.MetricFixtures.GetLatestMetricByKey(context.Background(), key)
+					s.Require().Nil(err)
+					s.Equal(iteration, lastMetric.LastIter)
+				}
+			}
+			if len(tt.latestMetricKeyCount) > 0 {
```

Comment:
I think you even don't need such `if` statement. `range` won't even iterate if `latestMetricKeyCount` is missing.

Reply:
removed `if`s